include/macroni/Translation/Macroni/MacroniVisitor.hpp

```diff
@@ -30,7 +31,7 @@ lowest_unvisited_substitution(pasta::Stmt &stmt,
 // macro parameters, if any.
 std::vector<llvm::StringRef> get_parameter_names(pasta::MacroSubstitution &sub);
 
-struct macroni_visitor : vast::cg::visitor_base {
+struct macroni_visitor : empty_visitor {
```

Comment:
What is the purpose of this visitor compared to the visitors provided already by vast (unsupp/killing one?).

Reply:
The point of this visitor is to provide default overrides for all `visitor_base`'s methods so that classes which inherit from the `empty_visitor` can just override the visit methods they are interested in overriding, and return `nullptr` for the other visit methods to tell the fallback visitor to try the next visitor instead.
I can't inherit from the unsupported visitor because then the inherited visit methods would return operations in the `unsup` dialect, which would tell the fallback visitor to stop "falling back" because they are not null values.
Likewise, I can't use the unreachable visitor because that would fail with a fatal error and exit the program.

Reply:
Okay, that makes sense, question is kinda why this isn't provided by some base class already in VAST. Keep it here though, the usage makes perfect sense.